src/Sylius/Bundle/ApiBundle/Filter/InclusiveDateRangeFilter.php

```diff
@@ -0,0 +1,89 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Sylius Sp. z o.o.
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sylius\Bundle\ApiBundle\Filter;
+
+use ApiPlatform\Core\Bridge\Doctrine\Orm\Filter\ContextAwareFilterInterface;
+use ApiPlatform\Core\Bridge\Doctrine\Orm\Util\QueryNameGeneratorInterface;
+use Doctrine\ORM\QueryBuilder;
+
+/**
+ * When given a date-only "before" value to a DateFilter, allow it to include dates on that day
+ *
+ * To include a '1970-01-01 13:00', it transforms:
+ *      'before' => '1970-01-01'
+ * into:
+ *      'before' => '1970-01-01 23:59'
+ */
+final class InclusiveDateRangeFilter implements ContextAwareFilterInterface
+{
+    /**
+     * @param ContextAwareFilterInterface $dateFilter
+     * @param string[] $properties
+     */
+    public function __construct(private ContextAwareFilterInterface $dateFilter, private array $properties)
+    {
+    }
+
+
+    /**
+     * @param class-string $resourceClass
+     * @return array<string, mixed>
+     */
+    public function getDescription(string $resourceClass): array
+    {
+        return $this->dateFilter->getDescription($resourceClass);
+    }
+
+    /**
+     * @param QueryBuilder $queryBuilder
+     * @param QueryNameGeneratorInterface $queryNameGenerator
+     * @param string $resourceClass
+     * @param string|null $operationName
+     * @param array<string, mixed> $context
+     * @return void
+     *
+     * @throws \Exception
+     */
+    public function apply(QueryBuilder $queryBuilder, QueryNameGeneratorInterface $queryNameGenerator, string $resourceClass, string $operationName = null, array $context = []): void
+    {
+        foreach ($this->properties as $property) {
+            $dateBefore = $context['filters'][$property]['before'] ?? null;
+            if ($this->isExclusive($dateBefore)) {
+                $context['filters'][$property]['before'] = $this->makeDateInclusive($dateBefore);
+            }
+        }
+
+        $this->dateFilter->apply($queryBuilder, $queryNameGenerator, $resourceClass, $operationName, $context);
+    }
+
+    protected function makeDateInclusive(string $value): string
+    {
+        return (new \DateTimeImmutable($value))
+            ->setTime(23, 59)
+            ->format('Y-m-d H:i');
+    }
+
+    private function isExclusive(mixed $value): bool
+    {
+        if(!$value || !is_string($value)) {
+            return false;
```

Comment:
Function named `isSomething()` shouldn't return false on invalid argument and throw exception instead.

I guess it's not a option for this case, so how about `$this->getDateTime(?string $date, string $defaultTime): ?string`

https://github.com/Sylius/Grid/blob/ca081e180bf1eaa25a7fc4e9493bd5b88a95a0cc/Filter/DateFilter.php#L67-L77

Reply:
I've pushed a new version of the Filter class, taking inspiration of the mentioned filter.
In the case of a Filter decorator, I didn't want to overstep or reproduce the base DateFilter validation & error handling, hence why I am not actively validating the item and just returning it as such.